python/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -315,13 +316,17 @@ class NearestNeighbors(Base,
         self.algorithm = algorithm
         self.algo_params = algo_params
         self.knn_index = <uintptr_t> 0
+        self.n_jobs = n_jobs
 
     @generate_docstring(X='dense_sparse')
     def fit(self, X, convert_dtype=True) -> "NearestNeighbors":
         """
         Fit GPU index for performing nearest neighbor queries.
 
         """
+        if self.n_jobs is not None:
+            warnings.warn("\nWarning: 'n_jobs' parameter is unsupported and only available for compatibility")
```

Comment:
The style checker is complaining that this line is longer than 80 characters. Besides that, this would create a lot of warnings when an automated caller creates estimators, no? cc @beckernick for opinion about warning

Reply:
Yes, I'd think that automated callers spawning child processes would cause this behavior, which we'd want to avoid (AutoGluon, Optuna, TPOT, etc.)

With that said, the intent of a warning is often to provide material information that is too important to risk being missed via documentation. Does this fit into that category? I could see this being simply a note in the docstring, like https://github.com/rapidsai/cuml/blob/903d5765abe40d0d29a75c470a36a9c2c1f8e927/python/cuml/ensemble/randomforestclassifier.pyx#L116-L124

Reply:
Would a simple warning such as `Warning: Nearest Neighbor can accept any named parameters (**kwargs), but they not all parameters will have an effect` be a sufficient addition to these notes here: https://github.com/rapidsai/cuml/blob/903d5765abe40d0d29a75c470a36a9c2c1f8e927/python/cuml/neighbors/nearest_neighbors.pyx#L276-L289

Reply:
A simple note in the docstring is more than good enough for this. 

The wording of any parameter is too general. Just a note that says that `'n_jobs' is accepted but with no effect in behavior for compatibility reasons`. The general behavior will be to only accept sklearn parameters probably, not any parameter in general.